azure-pipelines.yml

```diff
@@ -57,6 +62,18 @@ jobs:
   - script: bundle exec bin/ci/cucumber.rb
     displayName: "exec cucumber"
 
+  - task: Bash@3
+    inputs:
+      targetType: filePath
+      filePath: "./bin/ci/az-publish.sh"
+    env:
+      AZURE_STORAGE_ACCOUNT: $(AzureStorageAccount)
+      AZURE_STORAGE_KEY: $(AzureStorageKey)
+      AZURE_STORAGE_CONNECTION_STRING: $(AzureStorageConnectionString)
+      SOURCE_BRANCH: $(Build.SourceBranch)
+    displayName: "Publish to Azure Blob Storage"
+    condition: eq(variables['Agent.JobName'], variables['AzurePublishWhen'])
```

Comment:
Please use the following condition: `condition: and(succeeded(), eq(variables['Agent.JobName'], variables['AzurePublishWhen']), startsWith(variables['Build.SourceBranch'], 'master'))`